xray/core/alignment.py

```diff
@@ -103,9 +103,17 @@ def partial_align(*objects, **kwargs):
 def align_variables(variables, join='outer', copy=False):
     """Align all DataArrays in the provided dict, leaving other values alone.
     """
-    alignable = [k for k, v in variables.items() if hasattr(v, 'indexes')]
-    aligned = align(*[variables[a] for a in alignable], join=join, copy=copy)
+    from .dataarray import DataArray
+    from pandas.core.generic import NDFrame
+
     new_variables = OrderedDict(variables)
+    # if an item is a Series / DataFrame / Panel, try and wrap it in a constructor
+    new_variables.update(OrderedDict(
+        (k, DataArray(v)) for k, v in variables.items() if isinstance(v, NDFrame)
```

Comment:
I would rather use `(pd.Series, pd.DataFrame, pd.Panel)` in place of `NDFrame`, which I'm not sure is part of the public API.